### ISSUE TITLE
Add capacity to parse new custom line items

### ISSUE THREAD

**Issue by kwheelan** [closup/process-xbrl#83]

**Comment by lucakato** [comment]
@kwheelan Hi. Could you explain what this feature should do? what examples are there that I can use to test and work on this?

**Comment by kwheelan** [comment]
@lucakato Sure, I will add more info. In the meantime, could you finish resolving #82 (I've created pull request #84 to close that issue and #81)? This issue would be good to tackle after merging those changes

**Comment by lucakato** [comment]
@kwheelan will do

**Comment by kwheelan** [comment]
@lucakato Just added detail in the description. Please create a new branch, and when you're ready for me to take a look, create a pull request for this issue's branch into `dev` and add me as a reviewer to the PR. Thanks

**Comment by lucakato** [comment]
@kwheelan do you know which files you tested for the no concept tag issue? when I first tried it just with Ogemaw excel file the concept showed for the same number as yours. I still modified the python code for handling white space and would like to test it.
<img width="1496" alt="image" src="https://github.com/user-attachments/assets/4f1988e4-7a44-4cae-ad7c-9f6e445f9e32">

**Comment by kwheelan** [comment]
@lucakato I used the Ogemaw file you sent me a few weeks ago ([updated_ACFR_Ogemaw_template (2).xlsx](https://github.com/user-attachments/files/16513733/updated_ACFR_Ogemaw_template.2.xlsx)), but maybe Sarrah uploaded a more recent template since? Which one did you test here?

**Comment by lucakato** [comment]
@kwheelan I used the same ogemaw file, with the random number that was causing issues removed.
[updated_ACFR_Ogemaw_template.xlsx](https://github.com/user-attachments/files/16513804/updated_ACFR_Ogemaw_template.xlsx)

**Comment by kwheelan** [comment]
@lucakato Weird. Using your version of the file I can see the concept for the first row but the same issue persists lower in the table:

<img width="1438" alt="Screenshot 2024-08-06 at 1 34 04 PM" src="https://github.com/user-attachments/assets/a5ce1db8-c0f7-475a-afdf-9a6dd1609e45">

**Comment by lucakato** [comment]
@kwheelan ok I will look into it. Inspecting the html file it looks like this for that part:
`<ix:nonFraction contextRef="Ddate_parse_error_GovernmentalActivities" name="acfr:RevenueUsedForGeneralGovernmentServicesAdministration" unitRef="USD" id="StatementofActivities_C95" decimals="0" format="ixt:num-dot-decimal" >1,974,877</ix:nonFraction>`

which seems correct(?). I also see it in the Lookup table. Do you think it's some other issue?
<img width="1173" alt="image" src="https://github.com/user-attachments/assets/3fb41eda-15d4-4b62-8d26-9f550ed33b3c">

**Comment by kwheelan** [comment]
@lucakato 

Sorry I missed this comment earlier. Seems like there are 2 issues, and I haven't solved either. The first is that this tag should actually be `acfr:ExpensesForGeneralGovernmentServicesAdministration` not revenues. The next two cells should be revenue, but are expenses. This is an issue throughout the table, and probably is caused by a bug in the `process_cells()` method for the `StatementOfActivities` class (https://github.com/closup/process-xbrl/blob/e6f1f0703d702df1defb0f46dec6375a63f9b5ad/app/models/statment_of_activities.py#L23). I think the second issue is that General Government Administration doesn't seem to be in the ACFR taxonomy (found on [YETI](https://yeti1.corefiling.com/yeti/resources/yeti-gwt/Yeti.jsp#tax~(id~282*v~934)!net~(a~5538*l~1071)!lang~(code~en-us)!rg~(rg~29*p~14)).) This would explain why the concepts aren't showing in the viewer.

I'm going on vacation for the next week, but hopefully this gives you enough info to keep working on it. Marc may also be able to help confirm if the General Government Administration tags are in the taxonomy.

**Comment by lucakato** [comment]
@kwheelan thanks for the comment I didn't realize it should have been expenses! I'll look into it.

**Comment by lucakato** [comment]
@kwheelan @nfitz1 
I think I figured out where the issue is happening and I want to check if I'm correct. In the Ogemaw template (attached) that I received, if you go to 'Statement of Activities' it has 'acfr:RevenueUsedForGeneralGovernmentServicesAdministration' as a tag. Should this be included or removed from the spreadsheet?

What happens with the code is it sees the column 'Expenses' and then it runs this line in statement_of_activites.py and assigns the first tag for the xbrl_tag.
```Python
                if len(xbrl_list) == 1 or col_name == "expenses":
                    # first tag in list from relevant cell in column A in the excel sheet
                    xbrl_tag = xbrl_list[0]
```
<img width="855" alt="image" src="https://github.com/user-attachments/assets/7b606b21-acb2-41e6-8b4c-ab592290c0a9">

[updated_ACFR_Ogemaw_template.xlsx](https://github.com/user-attachments/files/16715947/updated_ACFR_Ogemaw_template.xlsx)

**Comment by kwheelan** [comment]
@lucakato You're right; the order is wrong on all the non-custom XBRL tags in that column. We do want to include `acfr:RevenueUsedForGeneralGovernmentServicesAdministration`, but the order is wrong. The list should be expense tag, revenue tag, net expense (revenue) tag.  The same issue happens with the Clayton example; I've just never noticed the swapped tags before.

So we just need to change the Excel function in column A for the Statement of Activities for rows 9+ to:
```
=IF(B14="", "Choose from drop-down -->", IF(COUNTIF('Lookup GovWide Stmt Activities'!$B:$B, B14) = 0, "acfr:ExpensesCustom, acfr:RevenueForOtherProgramsCustom, acfr:NetExpenseRevenueCustom", _xlfn.CONCAT(_xlfn.XLOOKUP(B14, 'Lookup GovWide Stmt Activities'!$B:$B, 'Lookup GovWide Stmt Activities'!$C:$C), ",", _xlfn.XLOOKUP(B14, 'Lookup GovWide Stmt Activities'!$B:$B, 'Lookup GovWide Stmt Activities'!$D:$D), ",", _xlfn.XLOOKUP(B14, 'Lookup GovWide Stmt Activities'!$B:$B, 'Lookup GovWide Stmt Activities'!$E:$E))))
```

(Basically just switch `'Lookup GovWide Stmt Activities'!$C:$C` and `'Lookup GovWide Stmt Activities'!$D:$D` to swap the position of the revenue and expense tags.)

I'm skeptical that this will fix all the parsing errors, but it should be a good start. Let me know how it works

**Comment by lucakato** [comment]
@kwheelan 
<img width="1499" alt="image" src="https://github.com/user-attachments/assets/184200ec-20e3-4051-943c-a7db9dab8b3f">
Do you mean like this?  Attached is my updated Ogemaw file:
Tried it but I still don't see the concept:
<img width="1225" alt="image" src="https://github.com/user-attachments/assets/c42acfd6-7be7-4fe7-8cd1-3343e3407903">
[updated_ACFR_Ogemaw_template.xlsx](https://github.com/user-attachments/files/16753894/updated_ACFR_Ogemaw_template.xlsx)

**Comment by kwheelan** [comment]
@lucakato Yes the order looks right in that cell (assuming the part of the cell that's not visible is the full expenses tag). 

But I think this only fixes part of the issue. The second issue is that General Government Administration doesn't seem to be in the ACFR taxonomy (found on [YETI](https://yeti1.corefiling.com/yeti/resources/yeti-gwt/Yeti.jsp#tax~(id~282*v~934)!net~(a~5538*l~1071)!lang~(code~en-us)!rg~(rg~29*p~14))). This would explain why the concepts aren't showing in the viewer.

The concepts (`RevenueUsedForGeneralGovernmentServicesAdministration` and `ExpensesForGeneralGovernmentServicesAdministration`) appear in the [Excel version of the taxonomy](https://docs.google.com/spreadsheets/d/18LhDW7fYq5cQaqxheBgqMBNHYAvxyOfO/edit?gid=1113124035#gid=1113124035) under `200000 - Statement - Activities - Expenses and Revenues for Programs` but not in YETI. Looking at the taxonomy in YETI, maybe the right tag is `acfr:ExpensesForGeneralGovernmentServices`? This is a question for @joffemd 

We'll want to update the Statement of Activities lookup table in Excel to match the tag we decide on here.

**Comment by joffemd** [comment]
@lucakato I will send this to my contact in India and ask him to create templates for two other cities so that we can do further testing.
